alacritty/src/display.rs

```diff
@@ -498,8 +525,31 @@ impl Display {
             });
         }
 
+        #[cfg(not(any(target_os = "macos", windows)))]
+        self.request_frame(&self.window);
+
         self.window.swap_buffers();
```

Comment:
Is there a reason why we do this before `swap_buffers`? Intuitively I would think that we'd ask for the next frame once the current frame has been sent off.

Reply:
It's required by protocol to be that way, since frame event will take effect only after commit. Read wl_surface::frame from https://github.com/alacritty/alacritty/blob/6b45780f3afa23e9e4277bf43e6bf381172fa1ea/alacritty.yml#L410 . Commit is done by swap_buffers internally here. If you swap them it won't work.